File: Programmers/java/LEVEL_1_[1차]_비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        List<String> map1 = new ArrayList<>();
        List<String> map2 = new ArrayList<>();

        for (int i = 0; i < arr1.length; i++) {
            map1.add(makeBinary(n, arr1[i]));
        }

        for (int i = 0; i < arr2.length; i++) {
            map2.add(makeBinary(n, arr2[i]));
        }

        for (int i = 0; i < n; i++) {
            String s = "";

            for (int j = 0; j < n; j++) {
                if (map1.get(i).charAt(j) == '1' || map2.get(i).charAt(j) == '1') {
                    s += '#';
                    continue;
                }

                s += ' ';
            }

            answer[i] = s;
        }

        return answer;
    }

    private String makeBinary(int limitLength, int n) {
        StringBuilder sb = new StringBuilder();

        while (n > 0) {
            if (n % 2 == 0) {
                sb.append('0');
            } else {
                sb.append('1');
            }

            n /= 2;
        }

        sb = sb.reverse();

        String s = sb.toString();
        while (s.length() < limitLength) {
            s = '0' + s;
        }

        return s;
    }
}

File: Programmers/cpp/LEVEL_1_[1차]_비밀지도.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string makeBinary(int limitLength, int n) {
	string s;

	while (n > 0) {
		if (n % 2 == 0) {
			s += '0';
		}
		else {
			s += '1';
		}

		n /= 2;
	}

	reverse(s.begin(), s.end());

	while (s.length() < limitLength) {
		s = '0' + s;
	}

	return s;
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
	vector<string> answer;

	vector<string> map1;
	vector<string> map2;

	for (int i = 0; i < arr1.size(); i++) {
		map1.push_back(makeBinary(n, arr1[i]));
	}

	for (int i = 0; i < arr2.size(); i++) {
		map2.push_back(makeBinary(n, arr2[i]));
	}

	for (int i = 0; i < n; i++) {
		string s;

		for (int j = 0; j < n; j++) {
			if (map1[i][j] == '1' || map2[i][j] == '1') {
				s += '#';
				continue;
			}

			s += ' ';
		}

		answer.push_back(s);
	}

	return answer;
}